main_withdc.py

```diff
@@ -0,0 +1,104 @@
+"""This runs main and outputs it to discord"""
+import os
+import sys
+import pickle
+import time
+import datetime
+import cv2
+import face_recognition
+import discord
+import functions
+
+CAMINDEX = 0
+SAVE_FILE = "encodings.pickle"
+FONT = cv2.FONT_HERSHEY_SIMPLEX
+FONTSCALE = 1
+TEXTCOLOUR = (255, 0, 0)
+TOLERANCE = 0.4
+if not os.path.exists(SAVE_FILE):
+    data = {"encodings":[],"names":[]}
+    print("No encoding file exists running face detection only")
+else:
+    with open(SAVE_FILE, "rb") as f:
+        data = pickle.load(f)
+    Knownencodings = data["encodings"]
+    names = data["names"]
+
+
+intents = discord.Intents.default()
+intents.message_content = True
+client = discord.Client(intents=intents)
+
+@client.event
+async def on_ready():# pylint: disable=too-many-locals,too-many-branches
+    ''' runs the face recognition here'''
+    prevmessage = ""
+    print(f"{client.user} logged in")
+
+    cap = functions.camera_connection(CAMINDEX)
+
+    while True:
+        ret , frame = cap.read()
+        if not ret:
+            print("couldnt get picture")
+            sys.exit(1)
+
+        boxes = face_recognition.face_locations(frame)
+        encodings = face_recognition.face_encodings(frame, boxes)
+        matches = []
+        foundnames = []
+        for facenum,encoding in enumerate(encodings):
```

Comment:
_optional goal due to time constraints:_ Extract this face scanning logic into a seperate recognition python script and call it from here.